receiver/carbonreceiver/protocol/delimiter_parser.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocol
+
+import (
+	"errors"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+)
+
+// DelimiterParser implements a parser that can breakdown the Carbon metric path
+// and transform it in corresponding metric labels.
+type DelimiterParser struct {
+	OrDemiliter string `mapstructure:"or_delimiter"`
+}
+
+var _ (Parser) = (*DelimiterParser)(nil)
+var _ (ParserConfig) = (*DelimiterParser)(nil)
+
+// BuildParser builds the respective parser of the configuration instance.
+func (d *DelimiterParser) BuildParser() (Parser, error) {
+	if d == nil {
+		return &DelimiterParser{}, nil
```

Comment:
Where is this code path executed? Do we have a case that uses this? Perhaps better ensure DelimiterParser is not nil at the call site? The way it is now is somewhat unconventional. I would not expect to be able to call a member function with nil receiver and for it to exhibit some default behavior. Is this idiomatic?

Reply:
This is happens in a few places, for instance gRPC does that with the GetXXX methods. In this case the config is the same as the actual parser structure. That said as the options for the delimiter are added it will make more sense to remove this code and avoid the question all together.

Reply:
I misread your comment on first pass, this is actually not actually needed. Removing.